app/controllers/posts_controller.rb

```diff
@@ -1,11 +1,12 @@
 class PostsController < ApplicationController
   def index
-    @user = User.find_by(id: params[:user_id])
-    @posts = @user.posts.order(id: :desc)
+    @user = User.find(params[:user_id])
+    @posts = @user.posts.includes(:user).where(user_id: params[:user_id]).references(:posts)
```

Comment:
- [OPTIONAL] As [required](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md) for this project, " `Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.` ", since you are fetching posts and comments for a particular user, your posts#index actions should use something like `@posts = @user.posts.includes(:comments)`. Please kindly fix that.

Reply:
Thanks for the suggestion